### PR TITLE
MAINT: Fix the typo "implment"

### PR DESCRIPTION
**Pull request by howjmay**
typo `implment` should be `implement`.

### CODE REVIEW

File: numpy/core/src/multiarray/convert_datatype.c [hunk 1]
```diff
@@ -1955,7 +1955,7 @@ PyArray_AddCastingImplementation_FromSpec(PyArrayMethod_Spec *spec, int private)
     if (meth == NULL) {
         return -1;
     }
-    int res = PyArray_AddCastingImplmentation(meth);
+    int res = PyArray_AddCastingimpelmentation(meth);
```
Comment (bashtage) [inline numpy/core/src/multiarray/convert_datatype.c:1958]: You shouldn't change the capitalization of this

File: numpy/core/src/multiarray/convert_datatype.h [hunk 2]
```diff
@@ -57,7 +57,7 @@ PyArray_FindConcatenationDescriptor(
         npy_intp n, PyArrayObject **arrays, PyObject *requested_dtype);
 
 NPY_NO_EXPORT int
-PyArray_AddCastingImplmentation(PyBoundArrayMethodObject *meth);
+PyArray_AddCastingimpelmentation(PyBoundArrayMethodObject *meth);
```
Comment (bashtage) [inline numpy/core/src/multiarray/convert_datatype.h:60]: Same here. You can fix spellign w/o changing the capitalization.

Comment (charris) [reply numpy/core/src/multiarray/convert_datatype.h:60]: It is still misspelled, should be "Implementation".

Comment (HowJMay) [reply numpy/core/src/multiarray/convert_datatype.h]: done. Thank you for reviewing

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
Thanks @HowJMay .

**Pull request merged**